powerauth-fido2/src/main/java/com/wultra/powerauth/fido2/service/AssertionService.java

```diff
@@ -75,12 +77,25 @@ public AssertionService(CryptographyService cryptographyService, AuthenticatorPr
      * @return Assertion challenge information.
      */
     public AssertionChallengeResponse requestAssertionChallenge(AssertionChallengeRequest request) throws Exception {
-        final AssertionChallenge assertionChallenge = assertionProvider.provideChallengeForAssertion(
-                request.getApplicationIds(), request.getTemplateName(), request.getParameters(), request.getExternalId()
-        );
+
+        final List<AuthenticatorDetail> authenticatorDetails = new ArrayList<>();
+
+        // If user ID is specified, fetch the user authenticators that should be allowed to respond the challenge
+        final String userId = request.getUserId();
+        if (userId != null) {
```

Comment:
We should move this logic to the provider. 
As of now:

- This service gets a list of possible authenticators from the provider. 
- Sends the list to the provider.
- The provider adds the list to his response.

Getting the list of possible authenticators and adding them to the response on the same level will increase readability.

Reply:
Didn't improve readability much, but OK